### DISCUSSION TITLE
Lesson 7: Integration test fails; no winner(?)

### DISCUSSION THREAD

**Discussion by tsopanid** [smartcontractkit/full-blockchain-solidity-course-py#88] [Q&A] [answered]
Hi,
I'm just doint the integration test, the unit tests passed. 
My output is:
```
=============================================================================================== FAILURES ===============================================================================================
_________________________________________________________________________________________ test_can_pick_winner _________________________________________________________________________________________

    def test_can_pick_winner():
        if network.show_active() in LOCAL_BLOCKCHAIN_ENVIRONMENTS:
            pytest.skip()
        lottery = deploy_lottery()
        account = get_account()
        lottery.startLottery({"from": account})
        lottery.enter({"from": account, "value": lottery.getEntranceFee()})
        lottery.enter({"from": account, "value": lottery.getEntranceFee()})
        fund_with_link(lottery)
        lottery.endLottery({"from": account})
        time.sleep(60)
>       assert lottery.recentWinner() == account
E       AssertionError: assert '0x0000000000...0000000000000' == <LocalAccount...0C6aBFd03F3d'>
E         +'0x0000000000000000000000000000000000000000'
E         -<LocalAccount '0x6D18D7028C011a678ff3220D64E90C6aBFd03F3d'>

tests/test_lottery_integration.py:23: AssertionError
======================================================================================= short test summary info ========================================================================================
FAILED tests/test_lottery_integration.py::test_can_pick_winner - AssertionError: assert '0x0000000000...0000000000000' == <LocalAccount...0C6aBFd03F3d'>
======================================================================== 1 failed, 1 skipped, 4 deselected in 173.53s (0:02:53) ========================================================================
```
I don`t fully nderstand what is going wrong. 
I checked my code and config, everything seems to be the same as in the repo.

I guess that somehow just the variable `recentWinner` is not saved.

When i dig in etherscan, i see an internal transaction with all the eth from the contract to my account. 
But why is it internal?

I see an event with the RequestedRandomness, so untill there everything seems to work.

I Wrote my own tests, they both pass:
```
def test_can_return_previous_randomness():
    if network.show_active() in LOCAL_BLOCKCHAIN_ENVIRONMENTS:
        pytest.skip()
    lottery = Lottery[-1]
    assert lottery.randomness() != 0


def test_can_return_recent_winner():
    if network.show_active() in LOCAL_BLOCKCHAIN_ENVIRONMENTS:
        pytest.skip()
    lottery = Lottery[-1]
    account = get_account()
    assert lottery.recentWinner() == account
```
Now i really have no idea why *test_can_pick_winner* doesn't pass...

**Comment by tsopanid** [comment, marked as answer]
Seems the *time.sleep(60)* was to short...
I tried with 180, now the test passed!

I learned a lot examining thi issue.
Having troubles sometimes can be fun :)

Thanx again for this great course!

**Reply by PatrickAlphaC** [reply]
Great work! Thanks for following up here!

**Reply by cherb-programs** [reply]
This worked for me too

**Comment by Programmusic** [comment]
**Here is a quick note for others reading this solution:** 

I had exactly the same assertionError on this test. I tried @tsopanid solution - thanks for sharing by the way, but it didn't work. I spent quite some time increasing the sleep time in various increments and waiting for longer and longer (!) for the test to run, and all failed.

So I went back and checked my code in some other places. The new element here was deploying to the Rinkeby testnet, as all unit test had passed. So I checked my config file and it turned out I had the wrong keyhash for Rinkeby.   Once I changed it to the same as the one in the Github repo it passed straight away - with sleep set to 60 as well.

So if you get this error at this stage try checking your settings for deployment to Rinkeby, including your  .env variables, as well as your config :) 

Thanks also to @PatrickAlphaC for making such a great course!

**Reply by PatrickAlphaC** [reply]
Awesome! Thank you so much for following up with your resolution here!
